test/controllers/class_controller_test.exs

```diff
@@ -275,4 +275,53 @@ defmodule CoursePlanner.ClassControllerTest do
 
     assert student_ids == attendance_student_ids
   end
+
+  test "hard deletes class and all attendances of it", %{conn: conn} do
+    course = insert(:course)
+    term1 = insert(:term, %{
+                            start_date: %Ecto.Date{day: 1, month: 1, year: 2017},
+                            end_date: %Ecto.Date{day: 1, month: 6, year: 2017},
+                            courses: [course]
+                           })
+
+    students = insert_list(3, :student)
+    offered_course = insert(:offered_course, %{term: term1, course: course, students: students})
+    class_attrs = %{@valid_attrs | offered_course_id: offered_course.id, status: "Planned"}
+    class = insert(:class, class_attrs)
```

Comment:
As I seek, the important parts here are the class, students and attendances.

I think we can define associations on the factories so we don't need to create everything here